### PR TITLE
CA1810: Report only if field belongs to ctor's type

### LINKED ISSUE #5784: false-positive for CA1810
### Analyzer

**Diagnostic ID**: [CA1810](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1810): `Initialize reference type static fields inline`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.100](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

False Positive. Static constructor sets static class fields.
(https://github.com/dotnet/msbuild/blob/797fd829a374a880f63fd4eea4ea6990404c48f4/src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs#L39-L49)

### Steps To Reproduce
Run the analyzer on the above mentioned code.

## Additional context
Using the analyzer on msbuild the related issue is [#7179](https://github.com/dotnet/msbuild/pull/7179)

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #5784

### CODE REVIEW

**Review by Youssef1313 — APPROVED** [review]

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InitializeStaticFieldsInlineTests.cs [hunk 1]
```diff
@@ -370,6 +370,41 @@ static C2()
...
+public class C1
+{
+    internal static int Size = 0;
+}
+
+public class C2
+{
+    static C2()
+    {
+        C1.Size = 42;
```
Comment (mavasani) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InitializeStaticFieldsInlineTests.cs:388]: What would be the result if the static ctor initializes both static fields of another type and also static fields of the containing type? Would it match our expected result in that case?

**Review by sharwell — CHANGES_REQUESTED** [review]
Haven't reviewed the implementation, but need to verify this is truly a false positive. CA1810 is a performance analyzer that I don't believe is enabled by default, so we'll want to make sure the analyzer is fully preserving the original intent after this change.

**Comment by Evangelink** [comment]
@sharwell Please let me know if you think that's better to close PR.

**Comment by sharwell** [comment]
@Evangelink yep, the analyzer is behaving by design here

**Pull request closed without merging**